### PR TITLE
feat: Don't calculate 100g nutriment if serving_quantity <=5 and value = '0', fixes #5848

### LINKED ISSUE #5848: Nutrition values for extra small serving (e.g. less than 1 g spray) cannot be extrapolated to 100g
### Is your feature request related to a problem? Please describe.

As of today (2021-10) we have 149 oils that should be Nutri-Score C, at most, but which are B (0.95% of the category). Nearly all these products are sprays. Producers are displaying nutrition facts per serving. As a serving represent less than 1g (spray), all the values are 0. Eg, https://world.openfoodfacts.org/product/0077890440834/olive-oil-cooking-spray-wegmans
This has bad effects:
- the products seem very healthy (!)
- the Nutri-Score is wrong
- we can't compare the products
- etc.

### Describe the solution you'd like

To be discussed
* Nutri-Score is not shown if all values are 0 and category is different from mineral waters, etc.? + alert to the user
* Nutri-Score not shown if product has values too different from category mean? + alert to the user
* Values not exported (json, api, CSV) if all values are 0 and category is different from mineral waters, etc.

### Number of products
* [Number of products with `serving_quantity` <= 10 g](https://mirabelle.openfoodfacts.org/products?sql=--+Products+with+a+data+quality+errors+and+serving_size+%0D%0Aselect+count+%28*%29+%0D%0A--code%2C+url%2C+serving_size%2C+serving_quantity%2C+data_quality_errors_tags%0D%0Afrom+%5Ball%5D%0D%0Awhere+%0D%0Aserving_size+%21%3D+%22%22%0D%0Aand+serving_quantity+%3C%3D+10%0D%0A--and+data_quality_errors_tags+%21%3D+%22%22%0D%0Aorder+by+random+%28%29%0D%0Alimit+100): ~47000 (2022-12)
* [Number of products with `serving_quantity` <= 10 g and a data quality error](https://mirabelle.openfoodfacts.org/products?sql=--+Products+with+a+data+quality+errors+and+serving_size+%0D%0Aselect+count+%28*%29+%0D%0A--%2C+code%2C+url%2C+serving_size%2C+serving_quantity%2C+data_quality_errors_tags%0D%0Afrom+%5Ball%5D%0D%0Awhere+%0D%0Aserving_size+%21%3D+%22%22%0D%0Aand+serving_quantity+%3C%3D+10%0D%0Aand+data_quality_errors_tags+%21%3D+%22%22%0D%0Alimit+100): 7100+ (2022-12)
* [Still 3965 products with `serving_quantity` <= 2 g](https://mirabelle.openfoodfacts.org/products?sql=--+Products+with+a+data+quality+errors+and+serving_size+%0D%0Aselect+count+%28*%29+%0D%0A--%2C+code%2C+url%2C+serving_size%2C+serving_quantity%2C+data_quality_errors_tags%0D%0Afrom+%5Ball%5D%0D%0Awhere+%0D%0Aserving_size+%21%3D+%22%22%0D%0Aand+serving_quantity+%3C%3D+2%0D%0Aand+data_quality_errors_tags+%21%3D+%22%22%0D%0Alimit+100)

### PR DESCRIPTION
**Pull request by john-gom**
### Related issue(s) and discussion

- Fixes #[ISSUE NUMBER]

### CODE REVIEW

File: lib/ProductOpener/Numbers.pm [hunk 1]
```diff
@@ -130,7 +130,12 @@ sub convert_string_to_number ($value) {
 	}
 	# Remove remaining commas that can be used as separators
 	$value =~ s/,//g;
-	$value += 0;
+
+	# If number has been recorded as 0.0 without decimals then want to preserve this
```
Comment (stephanegigandet) [inline lib/ProductOpener/Numbers.pm:134]: Today we don't have a way to store "4.0" or "0.0", they are converted to 4 and 0. The API expects numbers for both reading and writing (even though we do support strings for writing).

**Comment by john-gom** [comment]
> Today we don't have a way to store "4.0" or "0.0", they are converted to 4 and 0. The API expects numbers for both reading and writing (even though we do support strings for writing). 

Do you anticipate [problem,s with what I am doing here then?

**Comment by aleene** [comment]
> > Today we don't have a way to store "4.0" or "0.0", they are converted to 4 and 0. The API expects numbers for both reading and writing (even though we do support strings for writing). 

The number of digits might imply an accuracy of the value. So 0.5 implies an error of 0.05. And 0.00 an error of 0.005. A kind of implied accuracy.
We could interpret values of < 0.2 as a value of 0 with an error of 0.2. So a separate error-field would be useful.

**Review by stephanegigandet — APPROVED** [review]
Perfect, thank you!

The PR is still marked as "draft", can we merge it?

**Review by stephanegigandet — APPROVED** [review]
Thank you!

**Pull request merged**